### PR TITLE
cassandra-cpp-driver 2.15.0 (new formula)

### PR DESCRIPTION
**Pull request by mattmassicotte**
- [✅] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [✅] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [✅] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [✅] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [✅] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Just a note that a formula for this driver was attempted a while ago, but was abandoned.

https://github.com/Homebrew/legacy-homebrew/pull/36511

### CODE REVIEW

File: Formula/cassandra-cpp-driver.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  url "https://github.com/datastax/cpp-driver/archive/2.15.0.tar.gz"
+  sha256 "8cccffc7f3b7a8edc51c5c4344fd6a60d5a7beec27a1a3fd52742003a2675cb9"
+  head "https://github.com/datastax/cpp-driver.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "cmake" => :build
+  depends_on "libtool" => :build
+
```
Comment (dawidd6) [inline Formula/cassandra-cpp-driver.rb]: No empty line here.

File: Formula/cassandra-cpp-driver.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  depends_on "libuv"
+  depends_on "openssl@1.1"
+  uses_from_macos "zlib"
```
Comment (dawidd6) [inline Formula/cassandra-cpp-driver.rb]: Empty line between `uses_from_macos` and `depends_on`.

File: Formula/cassandra-cpp-driver.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  desc "DataStax C/C++ Driver for Apache Cassandra"
+  homepage "https://docs.datastax.com/en/developer/cpp-driver/latest"
+  url "https://github.com/datastax/cpp-driver/archive/2.15.0.tar.gz"
+  sha256 "8cccffc7f3b7a8edc51c5c4344fd6a60d5a7beec27a1a3fd52742003a2675cb9"
+  head "https://github.com/datastax/cpp-driver.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "cmake" => :build
```
Comment (dawidd6) [inline Formula/cassandra-cpp-driver.rb]: This line is duplicated.

Comment (mattmassicotte) [reply Formula/cassandra-cpp-driver.rb]: wow how did I do that

File: Formula/cassandra-cpp-driver.rb (hunk 2 above)
Comment (mattmassicotte) [reply Formula/cassandra-cpp-driver.rb]: done

File: Formula/cassandra-cpp-driver.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class CassandraCppDriver < Formula
+  desc "DataStax C/C++ Driver for Apache Cassandra"
+  homepage "https://docs.datastax.com/en/developer/cpp-driver/latest"
+  url "https://github.com/datastax/cpp-driver/archive/2.15.0.tar.gz"
+  sha256 "8cccffc7f3b7a8edc51c5c4344fd6a60d5a7beec27a1a3fd52742003a2675cb9"
+  head "https://github.com/datastax/cpp-driver.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
```
Comment (zbeekman) [inline Formula/cassandra-cpp-driver.rb]: Why in the world does this formula being built with CMake depend on autoconf, automake and libtool??!?

Comment (zbeekman) [reply Formula/cassandra-cpp-driver.rb]: Looking at the [documentation](https://docs.datastax.com/en/developer/cpp-driver/2.15/topics/building/#dependencies) it appears there is no need to add all this auto-tools stuff...

File: Formula/cassandra-cpp-driver.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+class CassandraCppDriver < Formula
+  desc "DataStax C/C++ Driver for Apache Cassandra"
+  homepage "https://docs.datastax.com/en/developer/cpp-driver/latest"
+  url "https://github.com/datastax/cpp-driver/archive/2.15.0.tar.gz"
+  sha256 "8cccffc7f3b7a8edc51c5c4344fd6a60d5a7beec27a1a3fd52742003a2675cb9"
+  head "https://github.com/datastax/cpp-driver.git"
+
+  depends_on "autoconf" => :build
```
Comment (zbeekman) [inline Formula/cassandra-cpp-driver.rb]:
```suggestion
```

File: Formula/cassandra-cpp-driver.rb (hunk 4 above)
Comment (zbeekman) [inline Formula/cassandra-cpp-driver.rb]:
```suggestion
 ```

File: Formula/cassandra-cpp-driver.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "libtool" => :build
```
Comment (zbeekman) [inline Formula/cassandra-cpp-driver.rb]:
```suggestion
```

**Review by zbeekman — CHANGES_REQUESTED** [review]

File: Formula/cassandra-cpp-driver.rb (hunk 6 above)
Comment (mattmassicotte) [reply Formula/cassandra-cpp-driver.rb]: Wow, thanks for looking at this. You clearly have a lot more familiarity with this kind of stuff than I do. I just duplicated exactly what the documentation said to do for building on macOS.

**Comment by mattmassicotte** [comment]
@zbeekman thank you for looking so closely. You were absolutely correct. I didn't even consider changing what the DataStax instructions said to do.

**Comment by zbeekman** [comment]
Could you improve the test to? Checking a pointer is not NULL is a pretty low bar. If possible a test could be copied/installed to the pkg_share directory from the upstream source.

**Comment by mattmassicotte** [comment]
I'm afraid I'm a little new to this. Thanks for the patience.

As written, the test validates that you can build and link successfully against the dylib, which should be enough for a successful install. The source repo has a large number of examples/tests, but all require a database server running to execute successfully. Do you happen to know of an similar formula that does something more like what you want?

**Comment by mattmassicotte** [comment]
I found an example in a Postgres driver that does a little more work. I’ll expand it to match. Good idea!

https://github.com/Homebrew/homebrew-core/blob/b752a6dd06757d71c778905df1ae838cd07d2fcc/Formula/libpq.rb

**Comment by mattmassicotte** [comment]
@zbeekman test updated! It now exercises more of the driver's API, and is very close to the test that libpq runs.

Also, I got in touch with the DataStax people to attempt to fix up their macOS build instructions. Thanks again for finding those issues!

File: Formula/cassandra-cpp-driver.rb [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        cass_future_free(future);
+
+        cass_session_free(session);
+        cass_cluster_free(cluster);
+
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-L#{lib}", "-lcassandra", "-o", "test"
+    assert_equal "connection failed", shell_output("./test")
```
Comment (zbeekman) [inline Formula/cassandra-cpp-driver.rb]: 💯

**Review by zbeekman — APPROVED** [review]
LGTM

**Comment by zbeekman** [comment]
Thanks for your contribution to Homebrew! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you @mattmassicotte!

**Pull request closed without merging**